client/src/components/Today/DetailedActivity.tsx

```diff
@@ -49,14 +61,113 @@ export default function DetailedActivity({
 				<span>{activity.name}</span>
 			</S.Title>
 
-			<S.EditButton
-				$color="blue"
-				onClick={(e) => {
-					e.stopPropagation();
-					openModal(modalIds.activities.form);
-				}}>
-				<PenLine size={20} />
-			</S.EditButton>
+			<div
+				style={{
+					position: "absolute",
+					top: "-1rem",
+					right: "6rem",
+					display: "flex",
+					flexDirection: "row",
+					gap: spacingValue.medium,
+					justifyContent: "flex-end",
+				}}
+			>
+				<Popover
+					withArrow
+					opened={opened}
+					onChange={toggle}
+					trapFocus
+					closeOnClickOutside
+				>
+					<Popover.Target>
+						<Tooltip
+							label="Recurring activities cannot be deleted yet"
+							withArrow
+							disabled={!activity.recurrence_id}
+						>
+							<Buttons.Action.Stylized
+								disabled={isNullish(activity.activity_id)}
+								$color="blue"
+								type="button"
+								onClick={(e) => {
+									e.stopPropagation();
+									toggle();
+								}}
+							>
+								<LucideX size={20} />
+							</Buttons.Action.Stylized>
+						</Tooltip>
+					</Popover.Target>
+					<Popover.Dropdown
+						style={{
+							boxShadow: "0 0.2rem 0.3rem -0.1rem #888",
+							backgroundColor: "#ddd",
+						}}
+					>
+						Delete this activity?
+						<Containers.Row
+							gap="small"
+							style={{ marginTop: spacingValue.smaller }}
+						>
+							<Buttons.Action.Default
+								$color="red"
+								type="button"
+								onClick={(e) => {
+									e.stopPropagation();
+									// prevents attempting to delete a synthetic activity
+									if (activity.activity_id) {
+										mutate(
+											{ activity_id: activity.activity_id },
+											{
+												onSuccess: () => {
+													close();
+													invalidateActivities();
+													// TODO (TRK-268) show a notification
```

Comment:
Need to integrate mantine notifications or expand my own notification thing significantly (which could be fun to implement).